test/matrix/dense_kernels.cpp

```diff
@@ -998,18 +998,203 @@ TEST_F(Dense, ExtractDiagonalOnShortFatIntoDenseCrossExecutor)
 }
 
 
+TEST_F(Dense, ComputeDotIsEquivalentToRef)
+{
+    set_up_vector_data(2);
```

Comment:
need () to test one vector

Reply:
good point, at least for (conj_)dot and norm2 we should test 1 rhs with stride == 1